tests/integration/targets/ec2_vpc_dhcp_option/tasks/main.yml

```diff
@@ -0,0 +1,863 @@
+---
+- module_defaults:
+    group/aws:
+      aws_access_key: "{{ aws_access_key }}"
+      aws_secret_key: "{{ aws_secret_key }}"
+      security_token: "{{ security_token | default('') }}"
+      region: "{{ aws_region }}"
+  vars:
+    vpc_cidr: '10.{{ 256 | random(seed=resource_prefix) }}.0.0/24'
+  block:
+
+  - name: create a VPC with a default DHCP option set to test inheritance and delete_old
+    ec2_vpc_net:
+      name: "{{ resource_prefix }}"
+      cidr_block: "{{ vpc_cidr }}"
+      state: present
+    register: vpc
+
+  - set_fact:
+      vpc_id: "{{ vpc.vpc.id }}"
+
+  - name: set up initial DHCP option set
+    environment:
+      AWS_ACCESS_KEY_ID: "{{ aws_access_key }}"
+      AWS_SECRET_ACCESS_KEY: "{{ aws_secret_key }}"
+      AWS_SESSION_TOKEN: "{{ security_token | default('') }}"
+      AWS_DEFAULT_REGION: "{{ aws_region }}"
+    block:
+      - name: find all DHCP options sets
+        command: 'aws ec2 describe-dhcp-options --filters --query "DhcpOptions[*].DhcpOptionsId" --output text'
+        register: associations
+
+      - name: find the associated VPCs
+        command: 'aws ec2 describe-vpcs --filters "Name=dhcp-options-id,Values={{ item }}" --query "Vpcs[*].VpcId" --output text'
+        loop: "{{ associations.stdout.split('\t') | list }}"
+        register: associated_vpcs
+
+      - set_fact:
+          in_use: True
+        when: vpc_id in item.stdout.split('\t') | list and item.stdout.split('\t') | list | length > 1
+        loop: "{{ associated_vpcs.results }}"
+
+      - name: create a new initial DHCP set if it already has another association
+        when: in_use is defined
+        block:
+        - command: 'aws ec2 associate-dhcp-options --vpc-id {{ vpc_id }} --dhcp-options-id default'
+
+        - name: create the new DHCP set
+          ec2_vpc_dhcp_option:
+            state: present
+            domain_name: "{{ (aws_region == 'us-east-1') | ternary('ec2.internal', aws_region + '.compute.internal') }}"
```

Comment:
Would we be better using something like "{{ resource_prefix }}.test" rather than a ternary?